### PR TITLE
Implemented /api/reviews endpoint

### LINKED ISSUE #60: Subissue: Backend 3 - create GET by userId endpoint for reviews
Creates a method in the controller (ReviewsController.java) to get all reviews for a given user.
The endpoint should be located at /api/reviews?userId=x
The user that is logged in should only be able to get their own issues. So
/api/reviews
Should only be performable when logged in as user x.

### PR DESCRIPTION
**Pull request by kev-yan**
closes #60 
This PR is part of the Epic: Dining Commons Reviews Backend.
In this PR, we implement the /reviews endpoint for accessing UCSB Reviews from the reviews API for the endpoint /reviews. If a logged in user calls this endpoint, it will return all reviews from the logged in user if there exists any reviews from the user.

In order for this change to be implemented, the name of the userId field needed to be changed from "user_id" from "userId" due to the inability to create repository function ".getByUser_id". Because this getter method wouldn't work, the userId field needed to be renamed in order for ".getByUserId" to work.

this is because the controller is checking if itemId exists in menuitems.

Dokku Deployment: https://proj-dining-review-backend-3.dokku-14.cs.ucsb.edu

### CODE REVIEW

**Review by Va1a — DISMISSED** [review]
LGTM

**Review by pconrad — CHANGES_REQUESTED** [review]
This PR appears to have many of the same problems as PR #55 so I don't see a point in rehashing all of those points here.

Fix PR55, and incorporate those changes into this PR, and make sure there is a thorough team code review that takes into account ALL of the prior feedback on PR55 and this one, as well as actually looking at the PR checklist.

**Review by amanpdesai — DISMISSED** [review]
LGTM

File: pom.xml [hunk 1]
```diff
@@ -165,6 +165,12 @@
       <version>4.29.2</version>
     </dependency>
 
+  
+    <dependency>
+      <groupId>org.apache.commons</groupId>
+      <artifactId>commons-lang3</artifactId>
+      <version>3.9</version>
+    </dependency>
```
Comment (pconrad) [inline pom.xml:173]: This appears to be to support:

```
import org.apache.commons.lang3.time.DateFormatUtils;
```

But it wasn't clear if you are actually using DateFormatUtils anywhere in this code.

So could you remove this change to the pom.xml as well as the unnecessary import?

File: src/main/java/edu/ucsb/cs156/dining/controllers/ReviewsController.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import org.springframework.security.web.bind.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.apache.commons.lang3.time.DateFormatUtils;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/dining/controllers/ReviewsController.java:35]: Is this import used anywhere?  If not, can it be removed?

File: src/main/java/edu/ucsb/cs156/dining/ExampleApplication.java [hunk 3]
```diff
@@ -14,6 +23,8 @@
  * The ExampleApplication class is the main entry point for the application.
  */
 @SpringBootApplication
+@EnableAsync  // @EnableJpaAuditing(dateTimeProviderRef = "utcDateTimeProvider") doesn't work with mvn tests but works with backend
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/dining/ExampleApplication.java:26]: Can we clean this up?

We don't put commented out code in PRs that are merged into main.

**Review by pconrad — CHANGES_REQUESTED** [review]

**Review by pconrad — CHANGES_REQUESTED** [review]
Can we remove the unnecessary changes to pom.xml, the unused import, and the commented out code?

**Review by Va1a — DISMISSED** [review]
CR changes addressed, lgtm.

**Review by Va1a — DISMISSED** [review]
Straggling comments removed, lgtm.

**Review by Va1a — APPROVED** [review]
merge conflicts fixed

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**